### ISSUE TITLE
Hardware Selection for OpenMpTCP

### ISSUE THREAD

**Issue by rifaterdemsahin** [Ysurac/openmptcprouter#1108]
Currently i use a skull canyon to host 5 connections and it is messy with usb adapters on esxi.

When i check the mhz it looks like no more than 400 Mhz when i process 400MBits/sec

Can I have the assumption that a device like this would work ok without issues as it is 1.4 Ghz till 1.4GBit/sec

https://www.amazon.co.uk/Kettop-Mi3865L-Untangle-Celeron-Gigabit/dp/B07FBZ87NJ/ref=sr_1_1?dchild=1&keywords=openwrt+intel&qid=1596132475&sr=8-1

How do i know if the device would work which openmptcp ?

**Comment by Ysurac** [comment]
Something like this should work and it have AES-NI support and this really help, but it's not really cheap ;)

**Comment by rifaterdemsahin** [comment]
`it's not really cheap ;)`
I agree my focus to is to improve the internet speed and stability

`AES-NI support`
The AES-NI instruction set extensions are used to optimize encryption and decryption algorithms on select Intel and AMD processors. Intel announced AES-NI in 2008 and released supported CPUs late 2010 with the Westmere architecture. AMD announced and shipped AES-NI support in 2010, starting with Bulldozer.

I am getting aggravate by esxi losing the network adapters on skull canyon. it is really not build to be a router.

I am comparing 
https://www.supermicro.com/en/products/chassis/1u/505/SC505-203B as a board
https://www.amazon.co.uk/gp/product/B01H6NE4X2/ref=ox_sc_act_title_4?smid=ADD8GDUWB40L7&psc=1 and get as a network card

and 
https://www.amazon.co.uk/Kettop-Mi3865L-Untangle-Celeron-Gigabit/dp/B07FBZ87NJ/ref=sr_1_1?dchild=1&keywords=openwrt+intel&qid=1596132475&sr=8-1

my goal is to get a sustained 600 MBits this year and have the infra to be able to increase speed by 30 percent every year on the same architecture. OpenMpTCP is very promising

**Comment by rifaterdemsahin** [comment]
I am also considering installing on Edgerouter 12p
https://eu.store.ui.com/products/edgerouter-12p

I dont know if i can

**Comment by gnosoz** [comment]
Guys, the main problem you are facing is based on the chipset speed of the
network card rather than processors speed / ram. Regardless of RJ45 port
number it all depends on the chipset processing the interrupts. separate
RJ45 cards gives overall better speed in my humble opinion and tests.
most routers and switches aren't built with MPCTP in mind and therefore the
less ports the better. More lan ports means less overall throughput, I am
basing this assumption on pure tests only.

the majority of cards for home are with 1.5gbps throughput altogether so if
you have a lot of traffic in the lan wan1 wan2 wan3 the total in any given
moment will never be greater than 1.5gbps.
if in this condition you have 3 fiber optics giving you 1gbps each in the
lan your max will still be 4-600Mbps given the fact you are using at the
same time the same chip for the wans.  this is pure physics I am afraid and
the only way out is to bypass that chipset adding separate network cards.
The USB can be another wayout but has the same principle as it all depends
on the usb chipset there...

**Comment by Ysurac** [comment]
Edgerouter 12p is not supported by OpenWrt, so it will not be supported by OpenMPTCProuter.

Processor speed is important for encryption, at least if there is no AES-NI support.
Else internal switch speed can be important if you want a big aggregated speed. There is now some with 2.5gbps (like BPI-R64, not yet supported).

**Comment by rifaterdemsahin** [comment]
`Edgerouter 12p is not supported by OpenWrt, so it will not be supported by OpenMPTCProuter.
`
bummer! Now i am thinking to use that device as the fallback router if OMR fails.

`Else internal switch speed can be important if you want a big aggregated speed. There is now some with 2.5gbps (like BPI-R64, not yet supported).
`
I am thinking to use 12p and connect to OMR with one cable and creating 5 virtual nics in ESXI which would end up in different wan connections. I don't know if i can I wanna try out.

it looks like this


![image](https://user-images.githubusercontent.com/5270036/89099111-e79c3600-d3e4-11ea-9fc1-22edc7e23378.png)

my rationale : I wanna be able to use OMR as the main gateway but when it fails i wanna fall back on the classic 12p ( sticky connections based action )

PS: Plugging to many usb ethernet cards is not a good idea for ESXI as it loses the connection at restarts.I wanna use the dedicated only one ethernet connection from the ESXI and divide it into 5 virtual ports where they would end up using only one connection from 12P. I am not sure that kind of forwarding could be setup using OMR and if it can detect the signals from the modem.( that part is not complete in my mind )

**Comment by rifaterdemsahin** [comment]
https://www.youtube.com/watch?v=UqUE1xiuKEY

> it looks like this

**Comment by jhackeur** [comment]
Great thread!

What do you think about this one:
https://fr.aliexpress.com/item/4000480680892.html?spm=a2g0o.productlist.0.0.4ef545a3xQVEwQ&algo_pvid=e4cf825a-597a-4946-a325-27438a1549c9&algo_expid=e4cf825a-597a-4946-a325-27438a1549c9-1&btsid=0b0a119a15984337175543505e7674&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_
160€

Should works ?

**Comment by Adorfer** [comment]
Concerning the "Kettop" (they are sold as e.g. "NRG Systems IPU661"), i'd go for an i3. 

As gnosoz pointed out: 

 >   Regardless of RJ45 port number it all depends on the chipset processing the interrupts. separate RJ45 cards gives overall better speed in my humble opinion and tests.

the layout of the Kettop is "one rj45 goes to one dedicated intel nic which is exclusivly asinged a PCIe lane", which sounds very promising. 
(e.g. if you like an IRQ/PCIe on the minipcie slot for wifi: requiring you to reflash the bios to disable eth6, since they do not like to share lanes/irqs)

https://www.amazon.co.uk/Kettop-Mi7100L6-Firewall-Support-Barebone/dp/B07L5QMW83

**Comment by jhackeur** [comment]
> thank you for the very unrefined, unsanitized url, splease be bit more specific what you mean by "this one"
> ![grafik](https://user-images.githubusercontent.com/10302493/91297601-defa0e00-e79e-11ea-8d3d-f0cfb63cbc47.png)

Indeed, sorry for the URL - solved.
I mean Pentuim 4405U with 4GB RAM

**Comment by Adorfer** [comment]
For evaluation i ordered this one. i will come back with results. 
![grafik](https://user-images.githubusercontent.com/10302493/91385693-c7b83080-e831-11ea-955e-311c38644f63.png)

**Comment by jhackeur** [comment]
> 
> 
> For evaluation i ordered this one. i will come back with results.
> ![grafik](https://user-images.githubusercontent.com/10302493/91385693-c7b83080-e831-11ea-955e-311c38644f63.png)

I'm looking forward for your feedback :)

**Comment by rifaterdemsahin** [comment]
i gave up working on my skull canyon the usb ethernet cards and ESXI is not stable so I am planning to buy
![image](https://user-images.githubusercontent.com/5270036/92244465-dd28fc80-eeba-11ea-8dfc-00cc40acfaac.png)

I dont know if this one would work
https://www.amazon.co.uk/gp/product/B076B6SWG5/ref=as_li_qf_asin_il_tl?ie=UTF8&tag=rifaterdemsah-21&creative=6738&linkCode=as2&creativeASIN=B076B6SWG5&linkId=60fd0178aa505784d9006093b8d000f1

**Comment by jhackeur** [comment]
> 
> 
> i gave up working on my skull canyon the usb ethernet cards and ESXI is not stable so I am planning to buy
> ![image](https://user-images.githubusercontent.com/5270036/92244465-dd28fc80-eeba-11ea-8dfc-00cc40acfaac.png)
> 
> I dont know if this one would work
> https://www.amazon.co.uk/gp/product/B076B6SWG5/ref=as_li_qf_asin_il_tl?ie=UTF8&tag=rifaterdemsah-21&creative=6738&linkCode=as2&creativeASIN=B076B6SWG5&linkId=60fd0178aa505784d9006093b8d000f1

Realy expansive compare to device founded on Aliexpress.
If device ordered by "Adorfer" works fine, that's the better option !

**Comment by Adorfer** [comment]
> 
> 
> For evaluation i ordered this one. i will come back with results.
> ![grafik](https://user-images.githubusercontent.com/10302493/91385693-c7b83080-e831-11ea-955e-311c38644f63.png)

not to keep you waiting for too long: 

  - unit arrived in good condition including decent PSU
  - there was a windows10 preinstalled, which boots within seconds due to EFI
  - bios is really good with all sorts of settings you would expect in a server board (redirect console/bios to COM-port etc)
  - install of OMR worked with a simple "dd" to the installed kingston-SSD-modules
  - LAN1-LAN5 recognized as eth0-eth4, 
  - no luck with LAN6 yet. documentation says "one PCIe-lane per interface", and this "last one" is reserverd for the pcie-slot (wifi....), there is an alternative bios to be flashed to use the lan6 (as far as i understood so far)

no time for a full setup so far, but at the moment it looks totally fine and working as expected.
i keep you posted. for benchmark. 

![grafik](https://user-images.githubusercontent.com/10302493/92738043-ea8f2c80-f37b-11ea-9d7a-333795862e01.png)

**Comment by rifaterdemsahin** [comment]
thanks mate what the max speed did you get combining ?

having windows on it can be a plus for the multi purpose usecase.

`install of OMR worked with a simple "dd" to the installed kingston-SSD-modules
`

this is great news. Any ali baba link for this device ?

**Comment by rifaterdemsahin** [comment]
price gets higher with a better core
![image](https://user-images.githubusercontent.com/5270036/93405212-4c6ffa80-f884-11ea-9e97-f5363853e06c.png)


I need to use this device and build a 2 GBit / sec network on top of it. Pentium would not cut it. ( Assumption )

**Comment by Adorfer** [comment]
Pentium4405U, 4GB RAM, 64GB SSD, 
3 WAN: 200/20, 50/10, 50/10. 
Getting alsways maximum speed on speedtests, a little shy of 300/40

CPU never coming out of idle: 
![grafik](https://user-images.githubusercontent.com/10302493/93476720-65b38e00-f8fa-11ea-937c-512ab3590f01.png)
RAM use... irrelevant
![grafik](https://user-images.githubusercontent.com/10302493/93476846-8aa80100-f8fa-11ea-9567-7a068eec87b3.png)
Filesystem: i have not resized the default-partition of 512MB... since: no need. 
![grafik](https://user-images.githubusercontent.com/10302493/93476989-bc20cc80-f8fa-11ea-8254-28bee0424d8f.png)

In other words: I do not want to discourage you from purchasing an i3 or even i5 with large SSD and several GB of RAM. 
For ORM on 1000TX LAN: Definitly not needed! thie 4405U would probably be able to handle more than 2GBit, if it had an interface for that or i would be able to figure out how to LAG towards the switch.

**Comment by openaspace** [comment]
> Pentium4405U, 4GB RAM, 64GB SSD,
> 3 WAN: 200/20, 50/10, 50/10.
> Getting alsways maximum speed on speedtests, a little shy of 300/40
> 
> CPU never coming really out of idle (you see the load during daytime hourse when there are people in the office. In plus there a constant upstream of about 12MBit/s 24/7 for a Cloud-CCTV system.)
> ![grafik](https://user-images.githubusercontent.com/10302493/93476720-65b38e00-f8fa-11ea-937c-512ab3590f01.png)
> RAM use... irrelevant
> ![grafik](https://user-images.githubusercontent.com/10302493/93476846-8aa80100-f8fa-11ea-9567-7a068eec87b3.png)
> Filesystem: i have not resized the default-partition of 512MB... since: no need.
> ![grafik](https://user-images.githubusercontent.com/10302493/93476989-bc20cc80-f8fa-11ea-8254-28bee0424d8f.png)
> 
> In other words: I do not want to discourage you from purchasing an i3 or even i5 with large SSD and several GB of RAM.
> For ORM on 1000TX LAN: Definitly not needed! thie 4405U would probably be able to handle more than 2GBit, if it had an interface for that or i would be able to figure out how to LAG towards the switch.

Hi,  a courtesy please, you can write me the exact motherboard model?
Really thank you.

**Comment by Adorfer** [comment]
no i can not, but here you have a photo from the device: 
![grafik](https://user-images.githubusercontent.com/10302493/95241459-0e744f80-080e-11eb-9aa2-17a5e4f555c6.png)

**Comment by rifaterdemsahin** [comment]
` For ORM on 1000TX LAN: Definitly not needed! thie 4405U would probably be able to handle more than 2GBit, if it had an interface for that or i would be able to figure out how to LAG towards the switch.`

I am considering  CPU: Intel Dual Core i5 7200U, 64 bit, up to 3.1GHz, 3MB Smart Cache, Intel AES-NI hardware support 

the reason is if the OMR does not work i could turn it into a pfsense box. I am focused on getting the best possible upload speeds to upload content to youtube.

**Comment by Adorfer** [comment]
just for note, i have now the 3rd of the is p4405u in use, this was a test on 800/50 + 100/25
![grafik](https://user-images.githubusercontent.com/10302493/99870118-bbddcf80-2bd0-11eb-8e48-b048bbd13024.png)

Idle with the supplied PSU is around 10W, on upload it peaks to 16W, 12W seems to be typical. 
![grafik](https://user-images.githubusercontent.com/10302493/99870130-dca62500-2bd0-11eb-80ca-c3b1d4ba022e.png)
